index.ts

```diff
@@ -32,17 +36,48 @@ server.tool(
     to: z.string().email().describe("Recipient email address"),
     subject: z.string().describe("Email subject line"),
     content: z.string().describe("Plain text email content"),
+    // If sender email address is not provided, the tool requires it as an argument
+    ...(!senderEmailAddress
+      ? {
+          from: z
+            .string()
+            .email()
+            .nonempty()
+            .describe(
+              "Sender email address. This must be provided explicitly provided by the end user and cannot be empty. Do not automatically populate this."
+            ),
+        }
+      : {}),
```

Comment:
Decided to only include the argument if `senderEmailAddress` is not defined, since it might confuse the agent/client of always including the `from` argument even though `senderEmailAddress` is defined